app/models/reservation.rb

```diff
@@ -13,6 +13,7 @@ class Reservation < ActiveRecord::Base
   #####
   belongs_to :product
   belongs_to :order_detail, inverse_of: :reservation
+  belongs_to :created_by_user, class_name: "User", foreign_key: :created_by
```

Comment:
I'm not a fan of this pattern, even though we use it elsewhere because `created_by` on some tables is really old and didn't originally have an association on it. So when we added the association, we went with `created_by_user` rather than changing the column name.

I'm partial to calling the column `created_by_id` or (actually probably better, but nothing matches this pattern) `created_by_user_id`, and then the association becomes `created_by`. We use a similar patter in `OrderDetails#dispute_by`, `Payment#paid_by`, `SangerSequencing::Batch#created_by`, `NuResearchSafety::Certificate#deleted_by`, and a few others.

It's one of those things where I much prefer the newer style, but we still have a bunch of the older style that it's somewhat inconsistent.

Thoughts?

Reply:
Ah, I originally had it as `created_by_id` but then moved away from that because I wasn't seeing that elsewhere in the codebase. I'll switch it.